### PR TITLE
ASAP:Int2019 60 automatic payment service

### PR DESCRIPTION
**Pull request by moleguaco99**

### CODE REVIEW

File: Server/VikingVault.Services/TimedPaymentService.cs [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+using VikingVault.DataAccess.Models;
+using VikingVault.Services.Abstractions;
+
+namespace VikingVault.Services
+{
+    public class TimedPaymentService : IHostedService
+    {
+        private Timer _timer;
+        public IServiceProvider Services { get; }
+        public int PaymentCheckInterval = 5 * 60;
```
Comment (Slitthe) [inline Server/VikingVault.Services/TimedPaymentService.cs]: Is this 300 what? 300 Spartans, sheep, seconds, minutes? Please add the unit of measurement as part of the variable name to make it descriptive

File: Server/VikingVault.Services/AdminService.cs [hunk 2]
```diff
@@ -12,6 +12,7 @@ namespace VikingVault.Services
 {
     public class AdminService : IAdminService
     {
+
```
Comment (robertpin) [inline Server/VikingVault.Services/AdminService.cs]: Remove empty line

File: Server/VikingVault.Services/ScopedProcessingService.cs [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using VikingVault.DataAccess;
+using VikingVault.DataAccess.Enums;
+using VikingVault.DataAccess.Models;
+
+namespace VikingVault.Services
+{
+    public interface IScopedProcessingService
```
Comment (crisfsabau) [inline Server/VikingVault.Services/ScopedProcessingService.cs]: Move in a separate .cs file

**Review by OlahCatalina — APPROVED** [review]

File: Server/VikingVault.Services/TimedPaymentService.cs [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+            {
+                var scopedProcessingService =
+                    scope.ServiceProvider
+                        .GetRequiredService<IScopedProcessingService>();
+
+                scopedProcessingService.PayCompanies();
+            }
+        }
+
+        public Task StopAsync(CancellationToken cancellationToken)
```
Comment (Slitthe) [inline Server/VikingVault.Services/TimedPaymentService.cs:44]: It doesn't seem to me like either async method `StopAsync`, `StartAsync` are doing any asynchronous work

Comment (moleguaco99) [reply Server/VikingVault.Services/TimedPaymentService.cs:44]: Yes, well these are overriden functions from IHostedService, so there is no way I can rename them

Comment (moleguaco99) [reply Server/VikingVault.Services/TimedPaymentService.cs:44]: Moreover, I cannot use another background service interface, I have to create a scope specifically for that type of service

**Pull request merged**